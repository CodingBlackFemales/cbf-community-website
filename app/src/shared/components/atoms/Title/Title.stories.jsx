import React from 'react';

import { storiesOf } from '@storybook/react';

import Title from '.';

storiesOf('Atoms', module).add('Title', () => <Title text={'Blog'} />);
