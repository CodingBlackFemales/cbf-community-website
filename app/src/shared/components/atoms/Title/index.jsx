import React from 'react';
import styled from 'styled-components';

const Title = ({ label, tagline = null, className }) => {
  const classPrefix = 'title';
  return (
    <div className={(classPrefix, className)}>
      <header className={`${classPrefix}__header`}>
        <h1 className={`${classPrefix}__label-text`}>{label}</h1>
        {tagline && <p className={`${classPrefix}__tagline-text`}>{tagline}</p>}
      </header>
    </div>
  );
};

const StyledTitle = styled(Title)`
  text-align: center;

  .title__header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .title__label-text {
    text-transform: uppercase;
    font-weight: 300;
    margin-bottom: 0.2em;
  }

  .title__tagline-text {
    margin-top: 0.2em;
    font-weight: 400;
  }
`;

export default StyledTitle;
