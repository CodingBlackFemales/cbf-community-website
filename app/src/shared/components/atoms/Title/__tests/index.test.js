import React from 'react';
import { cleanup } from '@testing-library/react';

import Title from '..';
import {
  mountWithTheme,
  renderWithTheme,
} from '../../../../utils/test/testHelpers';

describe('<Title', () => {
  afterEach(cleanup);
  const titleComponent = <Title label={'A title'} tagline={'A tagline'} />;
  const titleComponentNoTitle = <Title label={'A title'} />;

  it('matches the snapshot, when the label and tagline props exist', () => {
    const { asFragment } = renderWithTheme(titleComponent);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot, when the label prop exists and the tagline does not exist', () => {
    const { asFragment } = renderWithTheme(titleComponentNoTitle);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the header text and tagline', () => {
    const component = mountWithTheme(titleComponent);

    const labelText = component.find('.title__label-text');
    const taglineText = component.find('.title__tagline-text');

    expect(labelText.exists()).toBe(true);
    expect(taglineText.exists()).toBe(true);
  });

  it('displays the header text only', () => {
    const component = mountWithTheme(titleComponentNoTitle);

    const labelText = component.find('.title__label-text');
    const taglineText = component.find('.title__tagline-text');

    expect(labelText.exists()).toBe(true);
    expect(taglineText.exists()).toBe(false);
  });
});
