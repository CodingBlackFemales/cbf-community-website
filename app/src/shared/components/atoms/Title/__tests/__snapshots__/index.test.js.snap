// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title matches the snapshot, when the label and tagline props exist 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
}

.c0 .title__header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .title__label-text {
  text-transform: uppercase;
  font-weight: 300;
  margin-bottom: 0.2em;
}

.c0 .title__tagline-text {
  margin-top: 0.2em;
  font-weight: 400;
}

<div
    class="c0"
  >
    <header
      class="title__header"
    >
      <h1
        class="title__label-text"
      >
        A title
      </h1>
      <p
        class="title__tagline-text"
      >
        A tagline
      </p>
    </header>
  </div>
</DocumentFragment>
`;

exports[`<Title matches the snapshot, when the label prop exists and the tagline does not exist 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
}

.c0 .title__header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .title__label-text {
  text-transform: uppercase;
  font-weight: 300;
  margin-bottom: 0.2em;
}

.c0 .title__tagline-text {
  margin-top: 0.2em;
  font-weight: 400;
}

<div
    class="c0"
  >
    <header
      class="title__header"
    >
      <h1
        class="title__label-text"
      >
        A title
      </h1>
    </header>
  </div>
</DocumentFragment>
`;
